aeppl/transforms.py

```diff
@@ -133,6 +173,205 @@ def default_transform(fgraph: FunctionGraph, node: Node) -> Optional[List[Node]]
+        if (
+            ancestor_node.owner
+            and isinstance(ancestor_node.owner.op, RandomVariable)
+            and ancestor_node not in rv_map_feature.rv_values
+            and not getattr(ancestor_node.tag, "ignore_logprob", False)
+        )
+    ]
+
+    if len(ancestor_base_rvs) != 1:
+        raise NotImplementedError(
+            f"A logp could not be derived for the the {out_var} RV associated "
+            f"with the {rv_map_feature.rv_values[out_var]} value variable. "
+            f"{len(ancestor_base_rvs)} valid base RVs were found, but 1 is required."
+        )
+
+    base_rv = ancestor_base_rvs[0]
+
+    if base_rv.dtype.startswith("int"):
+        raise NotImplementedError(
+            f"A logprob cannot be derived for the transformed {base_rv}, because"
+            " it is discrete"
+        )
+
+    # Collect chain of nodes between base_rv and out_var (the derived RV)
+    rv_chain_nodes = [var.owner for var in vars_between_narrow(base_rv, out_var)[1:]]
+
+    invalid_ops = [
+        tr
+        for tr in rv_chain_nodes
+        if (
+            not isinstance(tr.op, implemented_transform_ops)
+            and not (
+                isinstance(tr.op, Elemwise)
+                and isinstance(tr.op.scalar_op, implemented_transform_ops)
+            )
+        )
+    ]
+    if invalid_ops:
+        raise NotImplementedError(
+            f"A logp could not be derived for the the {out_var} RV associated "
+            f"with the {rv_map_feature.rv_values[out_var]} value variable. "
+            f"The following OPs between {out_var} and {base_rv} cannot be "
+            f"inverted: {invalid_ops}."
+        )
+
+    # Create list of RV transforms that link the base_rv support to that of the
+    # value var / derived RV. If Y = at.exp(base_rv * 5), this should be
+    # [ScaleTransform(scale=5), ExpTransform()]
+    transform_inputs = list(base_rv.owner.inputs)
+    chain_transform_list = []
+    for transform_node in rv_chain_nodes:
+        transform_node_op = transform_node.op.scalar_op
+
+        if isinstance(transform_node_op, (Add, Mul)):
+            # Find non-rv branches
+            other_branches = [
+                inp
+                for inp in transform_node.inputs
+                if (inp.owner not in rv_chain_nodes and inp is not base_rv)
+            ]
+
+            if isinstance(transform_node_op, Add):
+                transform_class = LocTransform
+                new_input = (
+                    at.add(other_branches)
+                    if len(other_branches) > 1
+                    else other_branches[0]
+                )
+            else:
+                transform_class = ScaleTransform
+                new_input = (
+                    at.mul(other_branches)
+                    if len(other_branches) > 1
+                    else other_branches[0]
+                )
+
+            transform = transform_class(
+                transform_args_fn=lambda *inputs: inputs[-1],
```

Comment:
Not happy with this, I'll try again the `lambda rv: new_input`, but I think it was failing with multiple transforms (overwriting the `new_input`). The indexing works because it's always the same?